pandas/core/frame.py

```diff
@@ -616,9 +615,9 @@ def __init__(
                 if arr.ndim != 0:
                     raise ValueError("DataFrame constructor not properly called!")
 
-                values = cast_scalar_to_array(
-                    (len(index), len(columns)), data, dtype=dtype
-                )
+                arr = np.atleast_2d(arr)
+                shape = (len(index), len(columns))
+                values = np.tile(arr, shape)
```

Comment:
Potentially even simpler would be `values = np.full(shape, arr)` ?

Reply:
good idea, will update